### DISCUSSION TITLE
Requirements for GUI in Legendary

### DISCUSSION THREAD

**Discussion by pktiuk** [legendary-gl/legendary#196] [Ideas]
Hello,
I think it would be good to describe requirements for GUI in Legendary.  
It would not only clarify what it should be able to do, but it would also allow other developers helping with providing this feature. (Some developers have already done it independently [Rare](https://github.com/Dummerle/Rare) [Rare](https://github.com/CommandMC/Rare) [Heroic](https://github.com/flavioislima/HeroicGamesLauncher))

I think it would be good to describe things like:
- recommended library for GUI (if there is any recommended)
- describe location of assets
- how it should be launched (as a part of main executable `legendary --gui` or add another executable integrated with desktop environment `legendary-gui`)
- its functionalities (simple as possible, or allowing graphical editing configs etc)
- other things relevant for contributors willing to help

**Comment by derrod** [comment]
I haven't really had time yet to make a proper plan for how the GUI is going to be built. But it is probably going to involve fairly substantial changes across the entire project to clean up some of the mess that has grown over time + add the functionality needed for a GUI.

All I can say in regards to your bullet points so far:

 * GUI library needs to be cross-platform without too much of a hassle (read: not GTK)
 * Assets should be stored in another subfolder in the config directory
 * In order to keep the main legendary project lean and also avoid breakage with existing packages, the GUI will be a separate project over at https://github.com/legendary-gl/legendary-gui (and once it's ready the main legendary repo will likely also move to the org)
 * Very simple for a start, then more features as time allows. I'd also like to have basic friend list management for example
 * Not much I can say there yet, I want to make some changes that will make outside contributions easier and more usable though (contributing document, style enforcement, etc.)

**Comment by Dummerle** [comment]
What about Rare? It's a simple GUI for legendary and based on PyQt5. So it is cross-platform. At the moment are not so much features implemented. You can:
- login
- Launch and install Games
- Edit configuration for legendary and games
- Updates
It is not finished and has some bugs, but the most important functions work

**Reply by derrod** [reply]
I'm not going to rule out that I'll fork one of the existing GUIs to be the official one, I just haven't really looked at them so far.

**Reply by pktiuk** [reply]
@derrod  
You should also look at [Heroic](https://github.com/flavioislima/HeroicGamesLauncher) It seems to be a bit more mature than Rare, but it may be a bit more difficult to integrate it as a part of legendary.
